activerecord/lib/active_record/query_cache.rb

```diff
@@ -44,8 +44,9 @@ def self.install_executor_hooks(executor = ActiveSupport::Executor)
       executor.register_hook(self)
 
       executor.to_complete do
-        # FIXME: This should be skipped when env['rack.test']
-        ActiveRecord::Base.clear_active_connections!
+        unless ActiveRecord::Base.connection.transaction_open?
+          ActiveRecord::Base.clear_active_connections!
+        end
```

Comment:
@sgrif my code should not be deal with other thread's connections. It tries to deal with current thread's connections on multiple connection pools.
In other words it's about connections to different databases.